test/unit/Github/CreateReleaseTextViaKeepAChangelogTest.php

```diff
@@ -0,0 +1,204 @@
+use function unlink;
+
+class CreateReleaseTextViaKeepAChangelogTest extends TestCase
+{
+    public function testReportsCannotCreateReleaseTextIfChangelogFileIsMissing(): void
+    {
+        $repositoryPath = $this->createMockRepositoryWithChangelog(
+            self::INVALID_CHANGELOG,
+            'NOT-A-CHANGELOG.md'
+        );
+
+        self::assertFalse(
+            (new CreateReleaseTextViaKeepAChangelog())
+                ->canCreateReleaseText(
+                    $this->createMockMilestone(),
+                    RepositoryName::fromFullName('example/repo'),
+                    SemVerVersion::fromMilestoneName('1.0.0'),
+                    BranchName::fromName('1.0.x'),
+                    $repositoryPath
+                )
+        );
+    }
+
+    public function testReportsCannotCreateReleaseTextIfChangelogFileDoesNotContainVersion(): void
+    {
+        $repositoryPath = $this->createMockRepositoryWithChangelog(
+            self::INVALID_CHANGELOG,
+            'CHANGELOG.md'
+        );
+
+        self::assertFalse(
+            (new CreateReleaseTextViaKeepAChangelog())
+                ->canCreateReleaseText(
+                    $this->createMockMilestone(),
+                    RepositoryName::fromFullName('example/repo'),
+                    SemVerVersion::fromMilestoneName('1.0.0'),
+                    BranchName::fromName('1.0.x'),
+                    $repositoryPath
+                )
+        );
+    }
+
+    public function testReportsCanCreateReleaseWhenChangelogWithVersionExists(): void
+    {
+        $changelogContents = sprintf(self::READY_CHANGELOG, date('Y-m-d'));
+        $repositoryPath    = $this->createMockRepositoryWithChangelog(
+            $changelogContents,
+            'CHANGELOG.md'
+        );
+
+        self::assertTrue(
+            (new CreateReleaseTextViaKeepAChangelog())
+                ->canCreateReleaseText(
+                    $this->createMockMilestone(),
+                    RepositoryName::fromFullName('example/repo'),
+                    SemVerVersion::fromMilestoneName('1.0.0'),
+                    BranchName::fromName('1.0.x'),
+                    $repositoryPath
+                )
+        );
+    }
+
+    public function testExtractsReleaseTextViaChangelogFile(): void
+    {
+        $date              = date('Y-m-d');
+        $changelogContents = sprintf(self::READY_CHANGELOG, $date);
+        $repositoryPath    = $this->createMockRepositoryWithChangelog(
+            $changelogContents,
+            'CHANGELOG.md'
+        );
+
+        $expected = sprintf(<<< 'END'
+            ### Added
+            
+            - Everything.
+            
+            ### Changed
+            
+            - Nothing.
+            
+            ### Deprecated
+            
+            - Nothing.
+            
+            ### Removed
+            
+            - Nothing.
+            
+            ### Fixed
+            
+            - Nothing.
+            END, $date);
```

Comment:
Also, heredoc/nowdoc no longer require the end marker to be on a line of their own followed by a `;`, which makes this work.

Reply:
Oooh, didn't know! Awesome, but it does indeed confuse the heck out of the github PHP parsing :D